logstreams/src/test/java/io/camunda/zeebe/logstreams/util/TestEntry.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.logstreams.util;
+
+import io.camunda.zeebe.logstreams.log.LogAppendEntry;
+import io.camunda.zeebe.logstreams.log.LoggedEvent;
+import io.camunda.zeebe.protocol.impl.record.RecordMetadata;
+import io.camunda.zeebe.protocol.impl.record.UnifiedRecordValue;
+import io.camunda.zeebe.protocol.record.intent.Intent;
+import io.camunda.zeebe.util.ReflectUtil;
+import org.assertj.core.api.AbstractObjectAssert;
+import org.assertj.core.api.Assertions;
+
+public record TestEntry(
+    long key, int sourceIndex, UnifiedRecordValue recordValue, RecordMetadata recordMetadata)
+    implements LogAppendEntry {
+
+  public static LogAppendEntry ofDefaults() {
+    return new TestLogAppendEntryBuilder().build();
+  }
+
+  public static TestLogAppendEntryBuilder builder() {
+    return new TestLogAppendEntryBuilder();
+  }
+
+  public static LogAppendEntry ofKey(final long key) {
+    return new TestLogAppendEntryBuilder().withKey(key).build();
+  }
+
+  public static final class TestEntryAssert
```

Comment:
:upside_down_face: When writing custom assertions, you typically want to keep a continuous representation of the assertion info, so you typically don't call `Assertions.assertThat`, since that creates a completely new, separate assertion, divorced from the initial `actual` object. You'll use things like `objects`, `lists`, or other helpers. See `TopologyAssert` or `DirectoryAssert` as examples, or just peek into the standard assert classes provided by the library. Very soft suggestion and more of a comment than anything, since this is a very specific purpose assertion that will probably only be used in the logstream